nixos/modules/services/networking/doh-server.nix

```
@@ -0,0 +1,156 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.doh-server;
+  toml = pkgs.formats.toml { };
+  configFile = toml.generate "config.toml" cfg.settings;
+in
+{
+  options.services.doh-server = {
```

IMO the module name should be the package name here
```suggestion
  options.services.dns-over-https = {
```

---

This package (https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/dn/dns-over-https/package.nix) provides a server (doh-server) and a client (doh-client). Maybe this leads to two different services (and so modules), IMO.